test/e2e/preferences-persistence.test.ts

```diff
@@ -0,0 +1,69 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT-0
+import useBrowser from '@cloudscape-design/browser-test-tools/use-browser';
+import TablePageObject from './page/table-page-object';
+
+describe('Table preferences persistence', () => {
+  async function testPages({
+    browser,
+    pages,
+    columns,
+  }: {
+    browser: WebdriverIO.Browser;
+    pages: string[];
+    columns: number;
+  }) {
+    const page = new TablePageObject(browser);
+
+    for (const pageId of pages) {
```

Comment:
Doing for loops inside a single test-case is a bad pattern, because when the test fails, it will be hard to understand at which page it failed.

Recommended approach


```js
const pagesToTest = [...]

for(const pageId of pagesToTest) {
   describe(pageId, () => {
       test('Views that use the common Distributions table do not share the table preferences', () => {...});

       test('Views that use the common Instances table do not share the table preferences', () => {...});
   })
}
```

Reply:
Also, I am not sure this needs to be tested at all. Because the only way it fails, only if somebody edits the property names directly, which is a design decision, not a code bug

Reply:
Note that there is a need to visit multiple pages in each test in order to assert that applying preferences changes in one has no effect in another. Visiting only one page per test defeats the purpose of the test. How would that work in the approach above?

Reply:
> I see accidentally copy-pasting a key name to be a fair enough reason to have tests that prevent this

This test does not prevent copy-paste issues, because the new copy-pasted page will not be included into `pagesToTest`. There are ways to write a test actually capturing it, but I do not think it is worth engineering effort

Reply:
> Also, I am not sure this needs to be tested at all. Because the only way it fails, only if somebody edits the local storage names directly, which is a design decision, not a code bug

If a change is conscious and intended, then the tests can be changed accordingly. For us, it was quite clear that the keys should be per page because we were confused every time a view would load in the non-initial state, so I wanted to give this a bit of a "guardrail mechanism".

Reply:
I see your point now. On the other hand, as this PR is a clone of an internal change, I would prefer not introducing changes directly here unless there is something really outstanding.

Reply:
Done (note: I had to manually remove the test file anyway because the export copies without deleting)